src/ucp/core/ucp_mm.h

```diff
@@ -111,4 +111,10 @@ ucp_memh2uct(ucp_mem_h memh, ucp_md_index_t md_idx)
         status; \
     })
 
+#define ucs_foreach_md(_index, _md_map)                             \
+    for (_index = ucs_ffs64(_md_map);                               \
+         (_index < 64) && (UCS_BIT(_index) <= (uint64_t)(_md_map)); \
```

Comment:
@hoopoepg i think this doesn't work (infinite loop) when there is a single MD in the map

Reply:
there is another issue: on non intel platforms ffs64(0) returns non-zero value
fixing it